src/js/components/StatusBar.js

```diff
@@ -20,12 +20,21 @@ function stealPortion(barSizes, indexesLessThanThreshold) {
         maxIndex = index;
       }
     });
+    // Possible we have unassigned portion that we should steal from first
+    if (unassignedPortion > maxSize) {
+      maxSize = unassignedPortion;
+      maxIndex = -1;
+    }
 
     let currentGrowthIndex = indexesLessThanThreshold[0];
     let sizeToSteal = maxSize * GROW_RATIO;
     // Reassign the portion
     barSizes[currentGrowthIndex] += sizeToSteal;
-    barSizes[maxIndex] -= sizeToSteal;
+    if (maxIndex > -1) {
+      barSizes[maxIndex] -= sizeToSteal;
```

Comment:
This was not introduced by here, so this is unrelated to this PR, however, this function modifies barSizes. I think it would be much nicer if this function could return new segments based on what is given.